### PR TITLE
Fix/connect to firebase

### PR DESCRIPTION
**Pull request by Qoookie**
# Overview

This PR completes the migration of the app’s data layer to Firebase by updating all repositories to use the Firebase-backed implementation. Several backend logic issues surfaced during the integration and were corrected to ensure consistent and reliable communication with the new data source.

# What
Fully implemented the Firebase-backed versions of:
- MapRepositoryFirebase
- EventRepositoryFirebase
- OrganizationRepositoryFirebase

Updated all backend logic so every repository operation (create, read, update, delete) correctly interacts with Firebase.
Aligned all domain models and request flows with the Firestore schema and security rules.
Updated existing tests to ensure they run exclusively against the Firebase Emulator, not the real user database.

Cleaned up several inconsistencies in tests that caused instability in the CI.

# Why

- To finalize the migration of the entire data layer to Firebase, ensuring all features rely on the same consistent and production-ready backend.
- To fix issues caused by partial or outdated repository implementations that resulted in incorrect data handling or failed operations.
- To improve overall reliability and correctness of the app’s data flow, especially for features that depend heavily on backend interactions (events, maps, organizations).
- To make the test suite deterministic, isolated, and safe, avoiding any risk of writing to the shared production-like database.

To prepare a solid foundation for upcoming features (areas, replacements, event overview/editing, etc.).

# How
Rewrote each repository to use the Firebase implementations, fixing issues such as:

- inconsistent ID generation,
- asynchronous flow problems.
- Updated domain models to match the Firestore data structure.

Reviewed all existing tests and ensured they:
- connect only to the Firestore Emulator,
- never touch the real database.
- Addressed CI inconsistencies caused by flaky or non-isolated tests.

image

Liked with: [#239](https://github.com/swent-agendapp/agendapp/issues/239)
_This PR message has been genrated b AI_

### CODE REVIEW

File: app/src/test/java/com/android/sample/model/event/EventTest.kt [hunk 1]
```diff
@@ -74,6 +75,96 @@ class EventTest {
...
+            startDate = startLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            endDate = endLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            color = EventPalette.Green,
+            endRecurrence = endRecurrenceLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            recurrence = RecurrenceStatus.Weekly)
+    assertEquals(events.size, 5)
+  }
+
+  @Test
+  fun `create A Weekly Event For 4 Weeks + 1 days Create 4 Event`() {
```
Comment (Barde-E) [inline app/src/test/java/com/android/sample/model/event/EventTest.kt]: Same here : `... Create 5 Event`

File: app/src/test/java/com/android/sample/model/event/EventTest.kt [hunk 2]
```diff
@@ -74,6 +75,96 @@ class EventTest {
...
+            startDate = startLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            endDate = endLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            color = EventPalette.Green,
+            endRecurrence = endRecurrenceLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            recurrence = RecurrenceStatus.Weekly)
+    assertEquals(events.size, 5)
+  }
+
+  @Test
+  fun `create A Weekly Event For 4 Weeks - 1 days Create 3 Event`() {
```
Comment (Barde-E) [inline app/src/test/java/com/android/sample/model/event/EventTest.kt]: Same here : `... Create 4 Event` (very good edge-case coverage by the way !)

File: app/src/main/java/com/android/sample/model/organization/UserRepositoryFirebase.kt
[the remark this answers is not in the document]
Comment (Qoookie) [reply app/src/main/java/com/android/sample/model/organization/UserRepositoryFirebase.kt]: done in [25e3f7c](https://github.com/swent-agendapp/agendapp/pull/279/commits/25e3f7ca3f353a2586731f61e704d2cc48bb21f4)

File: app/src/test/java/com/android/sample/model/event/EventTest.kt [hunk 3]
```diff
@@ -74,6 +75,96 @@ class EventTest {
...
+            startDate = startLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            endDate = endLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            color = EventPalette.Green,
+            endRecurrence = endRecurrenceLocal.atZone(ZoneId.systemDefault()).toInstant(),
+            recurrence = RecurrenceStatus.Weekly)
+    assertEquals(events.size, 4)
+  }
+
+  @Test
+  fun `create A Monthly Event For 1 Year Create 12 Event`() {
```
[the remark this answers is not in the document]
Comment (Qoookie) [reply app/src/test/java/com/android/sample/model/event/EventTest.kt]: done in [25e3f7c](https://github.com/swent-agendapp/agendapp/pull/279/commits/25e3f7ca3f353a2586731f61e704d2cc48bb21f4)

File: app/src/test/java/com/android/sample/model/event/EventTest.kt (hunk 1 above)
Comment (Qoookie) [reply app/src/test/java/com/android/sample/model/event/EventTest.kt]: done in [25e3f7c](https://github.com/swent-agendapp/agendapp/pull/279/commits/25e3f7ca3f353a2586731f61e704d2cc48bb21f4)
